src/ProfileEndpoints.jl

```diff
@@ -117,17 +127,39 @@ function _start_cpu_profile(n, delay)
     return resp
 end
 
-function _cpu_profile_response(filename; with_pprof::Bool)
+function handle_cpu_profile_stop(qp, stage_path = nothing)
+    with_pprof = parse(Bool, get(qp, "pprof", default_pprof()))
+    filename = stage_path === nothing ? "cpu_profile" : joinpath(stage_path, "cpu_profile")
+    should_write_data = stage_path !== nothing
+    # Defer the potentially expensive profile symbolication to a non-interactive thread
+    return fetch(Threads.@spawn _cpu_profile_response($filename, $should_write_data; with_pprof=$with_pprof))
+end
+
+function _cpu_profile_response(filename, should_write_data = false; with_pprof::Bool)
```

Comment:
Same here, it's a bit weird that you're constructing a filename in the caller, even though it's not used. Maybe two methods here? Or just write to the filename in the first place instead of moving? Either way seems reasonable to me